### PR TITLE
Forward porting changes from azureml-rag

### PR DESCRIPTION
**Pull request by shpengmsft**
# Description

Forward porting following changes  from azureml-rag PR 1193855:

- Try use langchain VectorStore and fallback to vendor.
- Support azure-search-documents==11.4.0b11.
- OpenAI request dynamic batch size.

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by luigiw — APPROVED** [review]

File: sdk/ai/azure-ai-generative/azure/ai/generative/index/_mlindex.py [hunk 1]
```diff
@@ -241,16 +251,27 @@ def as_langchain_vectorstore(self, credential: Optional[TokenCredential] = None)
...
+
+                        try:
+                            from langchain.vectorstores import FAISS
+
+                            store = FAISS.load_local(str(tmpdir), embeddings)
+                        except Exception as e:
+                            logger.warning(
+                                f"Failed to load FAISS Index using installed version of langchain, retrying with vendored FAISS VectorStore.\n{e}"
+                            )
+                            from azureml.rag.langchain.vendor.vectorstores.faiss import FAISS
```
Comment (tot0) [inline sdk/ai/azure-ai-generative/azure/ai/generative/index/_mlindex.py:272]: This needs to import from azure.ai.generative: `from azure.ai.generative.index._langchain.vendor.vectorstores.faiss import FAISS`

File: sdk/ai/azure-ai-generative/azure/ai/generative/index/_mlindex.py [hunk 2]
```diff
@@ -626,21 +732,50 @@ def from_embeddings_container(
...
+
+            mlindex = create_index_from_raw_embeddings(
+                embeddings,
+                index_config,
+                connection=connection_args,
+                output_path=str(output_path),
+                credential=credential,
+            )
+        elif index_type == "pinecone":
+            from azureml.rag.tasks.update_pinecone import create_index_from_raw_embeddings
```
Comment (tot0) [inline sdk/ai/azure-ai-generative/azure/ai/generative/index/_mlindex.py:756]: This needs to import from azure.ai.generative: `azure.ai.generative.index._.tasks.update_pinecone`, also I don't think this task has been synced to azure-ai-generative yet, @datdo-msft is planning to do that tomorrow, so for now maybe leave our handling of pinecone indexes?

**Review by tot0 — COMMENTED** [review]
Couple of import fixes needed

File: sdk/ai/azure-ai-generative/azure/ai/generative/index/_mlindex.py (hunk 1 above)
Comment (shpengmsft) [reply sdk/ai/azure-ai-generative/azure/ai/generative/index/_mlindex.py:272]: Good catch. I did a quick search for "azureml.rag". will fix them all in next iteration.

File: sdk/ai/azure-ai-generative/azure/ai/generative/index/_mlindex.py (hunk 2 above)
Comment (shpengmsft) [reply sdk/ai/azure-ai-generative/azure/ai/generative/index/_mlindex.py:756]: I have to addi entire update_pinecone.py into this PR.

File: sdk/ai/azure-ai-generative/azure/ai/generative/index/_mlindex.py [hunk 3]
```diff
@@ -212,24 +221,25 @@ def as_langchain_vectorstore(self, credential: Optional[TokenCredential] = None)
                         index_name=self.index_config.get("index"),
                         embedding_function=self.get_langchain_embeddings(credential=credential).embed_query,
                         search_type="hybrid",
-                        semantic_configuration_name=self.index_config.get("semantic_configuration_name", "azureml-default"),
+                        semantic_configuration_name=self.index_config.get(
+                            "semantic_configuration_name", "azureml-default"
+                        ),
                         user_agent=f"azureml-rag=={version}/mlindex,langchain=={langchain_version}",
                     )
```
Comment (shpengmsft) [inline sdk/ai/azure-ai-generative/azure/ai/generative/index/_mlindex.py:228]: @tot0, Do you think we need to have a different user_agent string? i.e. "azure-ai-generative-index ==..."?

**Review by tot0 — APPROVED** [review]

**Pull request merged**